README.md
# Repository Coverage

[Full report](https://htmlpreview.github.io/?https://github.com/bashu/django-easy-maps/blob/python-coverage-comment-action-data/htmlcov/index.html)

| Name                                        |    Stmts |     Miss |   Cover |   Missing |
|-------------------------------------------- | -------: | -------: | ------: | --------: |
| easy\_maps/\_\_init\_\_.py                  |        0 |        0 |    100% |           |
| easy\_maps/admin.py                         |       25 |       25 |      0% |      1-45 |
| easy\_maps/conf.py                          |       12 |        0 |    100% |           |
| easy\_maps/geocode.py                       |       15 |        3 |     80% |     23-26 |
| easy\_maps/models.py                        |       58 |        5 |     91% |6-7, 27, 57, 73 |
| easy\_maps/templatetags/\_\_init\_\_.py     |        0 |        0 |    100% |           |
| easy\_maps/templatetags/easy\_maps\_tags.py |       38 |        6 |     84% |18, 55-60, 63-68, 77 |
| easy\_maps/utils.py                         |       24 |        5 |     79% |     29-33 |
| easy\_maps/widgets.py                       |       12 |       12 |      0% |      1-21 |
| **TOTAL**                                   |  **184** |   **56** | **70%** |           |


## Setup coverage badge

Below are examples of the badges you can use in your main branch `README` file.

### Direct image

[![Coverage badge](https://raw.githubusercontent.com/bashu/django-easy-maps/python-coverage-comment-action-data/badge.svg)](https://htmlpreview.github.io/?https://github.com/bashu/django-easy-maps/blob/python-coverage-comment-action-data/htmlcov/index.html)

This is the one to use if your repository is private or if you don't want to customize anything.

### [Shields.io](https://shields.io) Json Endpoint

[![Coverage badge](https://img.shields.io/endpoint?url=https://raw.githubusercontent.com/bashu/django-easy-maps/python-coverage-comment-action-data/endpoint.json)](https://htmlpreview.github.io/?https://github.com/bashu/django-easy-maps/blob/python-coverage-comment-action-data/htmlcov/index.html)

Using this one will allow you to [customize](https://shields.io/endpoint) the look of your badge.
It won't work with private repositories. It won't be refreshed more than once per five minutes.

### [Shields.io](https://shields.io) Dynamic Badge

[![Coverage badge](https://img.shields.io/badge/dynamic/json?color=brightgreen&label=coverage&query=%24.message&url=https%3A%2F%2Fraw.githubusercontent.com%2Fbashu%2Fdjango-easy-maps%2Fpython-coverage-comment-action-data%2Fendpoint.json)](https://htmlpreview.github.io/?https://github.com/bashu/django-easy-maps/blob/python-coverage-comment-action-data/htmlcov/index.html)

This one will always be the same color. It won't work for private repos. I'm not even sure why we included it.

## What is that?

This branch is part of the
[python-coverage-comment-action](https://github.com/marketplace/actions/python-coverage-comment)
GitHub Action. All the files in this branch are automatically generated and may be
overwritten at any moment.

CHANGES
1.1.3 (2021-10-09)
------------------

- Removed deprecaded code, various bugfixes.

1.1.2 (2020-10-02)
------------------

- Added support for Python 3.9

1.1.1 (2020-07-04)
------------------

- Dropped support for Python 2.x

1.1.0 (2020-02-29)
------------------

- Added support for Django 3.x.

1.0.2 (2019-05-28)
------------------

- Check is EASY_MAPS_GOOGLE_MAPS_API_KEY is not None before raising warning.

1.0.1 (2019-04-21)
------------------

- Fixed using callback for a non computed address

1.0.0 (2019-03-29)
------------------

- Added new option EASY_MAPS_ZOOM (16 by default).
- Added new option EASY_MAPS_LANGUAGE ('en' by default).
- EASY_MAPS_GOOGLE_MAPS_API_KEY deprecated in favor of historical
  EASY_MAPS_GOOGLE_KEY option.

0.9.4 (2019-03-28)
------------------

- Added support for Django 2.x, dropped support for Django < 1.11. It may
  still work with Django 1.8, but this is no longer tested.
- Make sure GoogleV3 geocoder respect API key.

0.9.3 (2016-11-11)
------------------

- Google Maps API key configuration.
- Revert from setuptools back to distutils.
- Russian translation is added.

0.9.2 (2015-07-12)
------------------

- Replacing broken 0.9.1 release, back to setuptools.

0.9.1 (2015-07-02)
------------------

- Resolve the 500 error when google send a no results info.
- Resolving width / height and other variables in template.

0.9 (2014-02-11)
----------------

- Backwards incompatible: added support for geopy >= 0.96,
  dropped support for geopy < 0.96.
- Added support for Django 1.6, dropped support for Django 1.3. It may
  still work with Django 1.3, but this is no longer tested.
- Experimental Python 3.3 support (no code changes - app seems to work as-is).

0.8.4 (2013-08-27)
------------------

- fix bad 0.8.3 release


0.8.3 (2013-08-27)
------------------

- ``easy_map`` tag now works when address is None.

0.8.2 (2013-07-02)
------------------

- Unique constraint is added to Address.address field (to prevent
  MultipleObjectsReturned exceptions).

  In order to upgrade, run

      python manage.py migrate easy_maps

- German translation is added.

0.8.1 (2013-03-25)
------------------

- Fix regressions in geocoding errors handling introduced in 0.8.

0.8 (2013-03-24)
----------------

- Testing improvements;
- EASY_MAPS_CENTER setting for default map coordinates;
- allow to pass an Address instance as argument of easy_map tag;
- better error handling;
- switch to GoogleV3 geocoder;
- customization hook: it is now possible to use a custom geocoding method;
- EASY_MAPS_GOOGLE_KEY now does nothing (it is not a meaningful option
  for V3 Geocoding API).

Minimum required Django version is 1.3 since this release.
It may work with older versions, but this is untested.

0.7.4 (2013-01-03)
------------------

- switch to https;
- make example settings Django 1.4 compatible;

0.7.3 (2012-09-21)
------------------

- use only first placemark from geocoder.

0.7.2 (2012-01-07)
------------------

- static fallback for map.html;
- fix localization of floats.

0.7.1 (2011-01-31)
------------------

- better error handling;
- EASY_MAPS_GOOGLE_KEY setting.

0.7 (2010-12-24)
----------------

- longtitude -> longitude;
- display is fixed for comma-delimited float locales.

0.6 (2010-12-02)
----------------

- admin preview widget;
- bugfixes.

0.5 (2010-12-01)
----------------

Initial release


example/settings.py
"""
Django settings for app project.

For more information on this file, see
https://docs.djangoproject.com/en/1.7/topics/settings/

For the full list of settings and their values, see
https://docs.djangoproject.com/en/1.7/ref/settings/
"""

# Build paths inside the project like this: os.path.join(BASE_DIR, ...)
import os

BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))


# Quick-start development settings - unsuitable for production
# See https://docs.djangoproject.com/en/1.7/howto/deployment/checklist/

# SECURITY WARNING: keep the secret key used in production secret!
SECRET_KEY = "YOUR_SECRET_KEY"

# SECURITY WARNING: don't run with debug turned on in production!
DEBUG = True

ALLOWED_HOSTS = []


# Application definition

PROJECT_APPS = [
    "easy_maps",
]

INSTALLED_APPS = [
    "django.contrib.admin",
    "django.contrib.auth",
    "django.contrib.contenttypes",
    "django.contrib.sessions",
    "django.contrib.messages",
    "django.contrib.staticfiles",
] + PROJECT_APPS

MIDDLEWARE_CLASSES = [
    "django.contrib.sessions.middleware.SessionMiddleware",
    "django.middleware.common.CommonMiddleware",
    "django.middleware.csrf.CsrfViewMiddleware",
    "django.contrib.auth.middleware.AuthenticationMiddleware",
    # "django.contrib.auth.middleware.SessionAuthenticationMiddleware",
    "django.contrib.messages.middleware.MessageMiddleware",
    "django.middleware.clickjacking.XFrameOptionsMiddleware",
    # "django.middleware.security.SecurityMiddleware",
]

MIDDLEWARE = MIDDLEWARE_CLASSES

ROOT_URLCONF = "example.urls"

SITE_ID = 1

TEMPLATES = [
    {
        "BACKEND": "django.template.backends.django.DjangoTemplates",
        "DIRS": [
            os.path.join(os.path.dirname(os.path.abspath(__file__)), "templates"),
        ],
        "APP_DIRS": True,
        "OPTIONS": {
            "context_processors": [
                "django.template.context_processors.debug",
                "django.template.context_processors.request",
                "django.contrib.auth.context_processors.auth",
                "django.contrib.messages.context_processors.messages",
            ],
        },
    },
]


# Database
# https://docs.djangoproject.com/en/1.7/ref/settings/#databases

DATABASES = {
    "default": {
        "ENGINE": "django.db.backends.sqlite3",
        "NAME": os.path.join(BASE_DIR, "db.sqlite3"),
    }
}


# Internationalization
# https://docs.djangoproject.com/en/1.7/topics/i18n/

LANGUAGE_CODE = "en-us"

TIME_ZONE = "UTC"

USE_I18N = True

USE_L10N = True

USE_TZ = True


# Static files (CSS, JavaScript, Images)
# https://docs.djangoproject.com/en/1.8/howto/static-files/

# Absolute filesystem path to the directory that will hold user-uploaded files.
MEDIA_ROOT = os.path.join(BASE_DIR, "media")

# URL that handles the media served from MEDIA_ROOT. Make sure to use a
# trailing slash.
MEDIA_URL = "/media/"

STATIC_ROOT = os.path.join(BASE_DIR, "staticfiles")

STATIC_URL = "/static/"


## Easy Maps settings

EASY_MAPS_GOOGLE_KEY = "AIzaSyATg_isuGSCHIlJamrxAXfkFDTYhIz7ytM"
EASY_MAPS_ZOOM = 16
EASY_MAPS_LANGUAGE = "ru"
